docs/sts/assume-role.md

```diff
@@ -0,0 +1,97 @@
+## AssumeRole [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+Returns a set of temporary security credentials that you can use to access Minio resources. AssumeRole requires authorization credentials for an existing user on Minio. The advantages of this API are
+
+- To be able to reliably use S3 multipart APIs feature of the SDKs without re-inventing the wheel of pre-signing the each URL in multipart API. This is very tedious to implement with all the scenarios of fault tolerance that's already implemented by the client SDK. The general client SDKs don't support multipart with presigned URLs.
+- To be able to easily get the temporary credentials to upload to a prefix. Make it possible for a client to upload a whole folder using the session. The server side applications need not create a presigned URL and serve to the client for each file. Since, the client would have the session it can do it by itself.
+
```

Comment:
Once can create another account on Minio for their temporary upload requirements. I am not sure quite sure what is the added benefit of AssumeRole API. This implementation does not implement adding another policy that further reduces what the entity that is assuming the role can do. Even temporary credentials can be done via adding a new user with restricted perms and then deleting it.

Reply:
Yes but that is not application requirement. The advantage of temp credentials is auto expiry and not shipping static creds to client apps which require using standard SDKs.

It is well known that using presigned URLs is complicated and users tend to prefer this API instead.

The role for this temp credentials is borrowed from an actual user, a role here is policy inherited. Now we can definitely add one more policy addition to restrict further. But I generally felt that is just an add on feature which can be implemented any time.

Using admin API is custom to Minio.. that is why we did external identity to have rotating creds. The most important thing in this PR is that even clients which require admin creds to be rotated and expired is possible. They do not have to ship admin creds everywhere.

Reply:
Ok, so we need to disable admin user calling assume role (like AWS).

Reply:
Done